src/gf_ring.h

```diff
@@ -324,10 +324,10 @@ T RingModN<T>::exp_quick(T base, T exponent) const
 
     T tmp = this->exp_quick(base, exponent / 2);
     // std::cout << base << "^" << exponent / 2 << "=" << tmp << "\n";
-    result = this->mul(tmp, tmp);
+    result = RingModN<T>::mul(tmp, tmp);
```

Comment:
This look like a bad fix, it break the tests…

Reply:
@lamphamsy any idea on how to fix this?

On one hand it seems that we must call the child's implementation, but on the other hand calling a virtual function in a constructor is really really error-prone (because the object is not fully initialized).